protomap/proto.go

```diff
@@ -359,3 +359,77 @@ func resolvedPath(basePath, annotatedPath *gpb.Path) *gpb.Path {
 	np.Elem = append(np.Elem, annotatedPath.Elem[len(basePath.Elem):]...)
 	return np
 }
+
+// ProtoFromPaths takes an input proto.Message and adds the values that are specified in the map vals to
+// it, using basePath as the prefix to any paths within the vals map. The message, p, is modified in place.
+// The map, vals, is keyed by the gNMI path to the field which is annotated in the ygot generated protobuf,
+// the complete path is taken to be basePath + the key found in the map.
+func ProtoFromPaths(p proto.Message, vals map[*gpb.Path]interface{}, basePath *gpb.Path) error {
+	m := p.ProtoReflect()
+
+	directCh := map[*gpb.Path]interface{}{}
+	for p, v := range vals {
+		// TODO(robjs): needs fixing for compressed schemas.
+		l := 0
+		if basePath != nil {
+			l = len(basePath.Elem)
+		}
+		if len(p.Elem) == l+1 {
+			directCh[p] = v
+		}
+	}
+
+	var rangeErr error
+	unpopRange{m}.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
+		annotatedPath, err := annotatedSchemaPath(fd)
+		if err != nil {
+			rangeErr = err
+			return false
+		}
+
+		for _, ap := range annotatedPath {
+			for chp, chv := range directCh {
+				if proto.Equal(ap, chp) {
+					if fd.Kind() == protoreflect.MessageKind {
```

Comment:
What should happen if there is a child that can't be found in the message? Should we error out?

Reply:
I see in the next PR you address this.